File: ckgame-java/src/main/java/com/ckgame/politics/CouncilPosition.java
package com.ckgame.politics;

import java.util.List;

/**
 * 内阁职位。
 */
public enum CouncilPosition {
    CHANCELLOR("首相"),
    MARSHAL("元帅"),
    STEWARD("总管"),
    SPYMASTER("间谍总管"),
    COURT_CHAPLAIN("宫廷神甫");

    private final String nameZh;

    CouncilPosition(String nameZh) {
        this.nameZh = nameZh;
    }

    public String nameZh() {
        return nameZh;
    }

    /** 返回所有职位。 */
    public static List<CouncilPosition> all() {
        return List.of(values());
    }
}

File: ckgame-ts/src/politics/CouncilPosition.ts
/**
 * Council positions.
 */
export enum CouncilPosition {
  CHANCELLOR = 'CHANCELLOR',
  MARSHAL = 'MARSHAL',
  STEWARD = 'STEWARD',
  SPYMASTER = 'SPYMASTER',
  COURT_CHAPLAIN = 'COURT_CHAPLAIN',
}

const councilPositionNames: Record<CouncilPosition, string> = {
  [CouncilPosition.CHANCELLOR]: '首相',
  [CouncilPosition.MARSHAL]: '元帅',
  [CouncilPosition.STEWARD]: '总管',
  [CouncilPosition.SPYMASTER]: '间谍总管',
  [CouncilPosition.COURT_CHAPLAIN]: '宫廷神甫',
};

export function councilPositionNameZh(pos: CouncilPosition): string {
  return councilPositionNames[pos] ?? pos;
}

/** All council positions. */
export function allCouncilPositions(): CouncilPosition[] {
  return [
    CouncilPosition.CHANCELLOR,
    CouncilPosition.MARSHAL,
    CouncilPosition.STEWARD,
    CouncilPosition.SPYMASTER,
    CouncilPosition.COURT_CHAPLAIN,
  ];
}
